### ISSUE TITLE
FullscreenImageCarousel accessibility

### ISSUE THREAD

**Issue by qroll** [LifeSG/react-design-system#888]
**Component**
`FullscreenImageCarousel`

**Describe the feature request**
- Align the FullscreenImageCarousel with the accessibility behaviour defined in the Figma specs

**Acceptance criteria**
- Roles should be correctly read out by the AT
- Focus should be trapped when the carousel overlay is active
- Can be navigated with the keyboard
- Minimally test in NVDA + Chrome (Windows), VO + Safari (macOS), Talkback + Chrome (Android)

**References**
[Figma](https://www.figma.com/design/wDAxbRmlq4xOn4UJ8zPuEF/%F0%9F%A7%91%F0%9F%8F%BB%E2%80%8D%F0%9F%A6%AF-Flagship-A11y-Annotations?node-id=2951-780&p=f&m=dev)
